lib/Basics/Exceptions.h

```diff
@@ -58,6 +58,25 @@
 #define THROW_ARANGO_EXCEPTION_MESSAGE(code, message) \
   throw arangodb::basics::Exception(code, message, __FILE__, __LINE__)
 
+/// @brief throws an arango result if the result fails
+#define THROW_ARANGO_EXCEPTION_IF_FAIL(reSUlt) \
+  do { \
+    if ((reSUlt).fail()) { \
+      throw arangodb::basics::Exception((reSUlt), __FILE__, __LINE__); \
+    } \
+  } while(0);
+
+#define CATCH_TO_RESULT_WITH_CODE(result, errorCode) \
+  catch (arangodb::basics::Exception const& e) { \
+    (result).reset(e.code(),e.message()); \
+  } catch (std::exception const& e) { \
+    (result).reset((errorCode),e.what()); \
+  } catch (...) { \
+    (result).reset((errorCode)); \
+  }
+
+#define CATCH_TO_RESULT(result) CATCH_TO_RESULT_WITH_CODE((result),TRI_ERROR_INTERNAL)
```

Comment:
Can we rename macro `CATCH_TO_RESULT_WITH_CODE` to just `CATCH_RESULT`, and remove macro `CATCH_TO_RESULT` altogether? I don't think we need two macros.

Reply:
Sure that could be done - It is probably better to force people to think about the correct default error code.